### PR TITLE
[Feat] Futurepass `delegate` examples

### PR DESCRIPTION
**Pull request by aidan-starke**
## Summary
 - [TRN-183](https://centralitydev.atlassian.net/browse/TRN-183)
 - Add `registerDelegate` example
 - Add `unregisterDelegate` example

## Checklist

- [x] Add description
- [x] Tag related issue(s)

### CODE REVIEW

File: examples/substrate/use-futurepass/src/registerDelegate.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  Governance = 3,
+  Staking = 4,
+}
+
+export async function main() {
+  const api = await getChainApi("porcini");
+  const caller = createKeyring(env.CALLER_PRIVATE_KEY);
+  const wallet = new Wallet(
+    env.CALLER_PRIVATE_KEY,
+    getDefaultProvider("https://porcini.rootnet.app/")
```
Comment (ken-centrality) [inline examples/substrate/use-futurepass/src/registerDelegate.ts:25]: use `getPublicProvider("porcini").provider`

File: examples/substrate/use-futurepass/src/registerDelegate.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  );
+
+  const deadline = 10000;
```
Comment (ken-centrality) [inline examples/substrate/use-futurepass/src/registerDelegate.ts:28]:
```suggestion
  const deadline = 75; // recommended low number, 75 blocks = 5 minutes
```

**Review by ken-centrality — COMMENTED** [review]
assume this is working for you when testing?

File: examples/substrate/use-futurepass/src/registerDelegate.ts (hunk 1 above)
Comment (aidan-starke) [reply examples/substrate/use-futurepass/src/registerDelegate.ts:25]: That is not an ethers provider

File: examples/substrate/use-futurepass/src/registerDelegate.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    await api.query.futurepass.holders(caller.address)
+  ).toString();
+
+  const message = ethers
+    .solidityKeccak256(
+      ["address", "address", "uint8", "uint32"],
+      [futurepass, delegate, proxyType, deadline]
+    )
+    .substring(2);
+  const encodedMessage = ethers.hexlify(ethers.toUtf8Bytes(message));
```
Comment (KarishmaBothara) [inline examples/substrate/use-futurepass/src/registerDelegate.ts:41]: We need to do this only when signing via metamask..
when we sign via wallet here.. we can skip ethers.hexlify(ethers.toUtf8Bytes(message))

https://github.com/futureversecom/trn-seed/blob/f991c529fc3c71df6379562274a44cfc445ef2d4/e2e/test/Futurepass/FuturepassPrecompile.test.ts#L156

Comment (aidan-starke) [reply examples/substrate/use-futurepass/src/registerDelegate.ts]: Cheers, though I am hitting [this error](https://github.com/futureversecom/trn-seed/blob/main/pallet/futurepass/src/lib.rs#L347) with either version 🤔

File: examples/substrate/use-futurepass/src/registerDelegate.ts (hunk 1 above)
Comment (ken-centrality) [reply examples/substrate/use-futurepass/src/registerDelegate.ts:26]: try update @therootnetwork/api to `1.0.2` and use `getPublicProviderUrl('porcini')`

Comment (aidan-starke) [reply examples/substrate/use-futurepass/src/registerDelegate.ts]: sorted 👌

File: examples/substrate/use-futurepass/src/registerDelegate.ts (hunk 3 above)
Comment (KarishmaBothara) [reply examples/substrate/use-futurepass/src/registerDelegate.ts]: did u manage to solve this?

File: examples/substrate/use-futurepass/src/registerDelegate.ts (hunk 2 above)
Comment (KarishmaBothara) [reply examples/substrate/use-futurepass/src/registerDelegate.ts]: Do we want to mention in README, why we recommend to have a short deadline?

File: examples/substrate/use-futurepass/src/registerDelegate.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  Governance = 3,
+  Staking = 4,
+}
+
+export async function main() {
+  const api = await getChainApi("porcini");
+  const caller = createKeyring(env.CALLER_PRIVATE_KEY);
+
+  const delegateWallet = new Wallet(
+    env.DELEGATE_PRIVATE_KEY,
```
Comment (ken-centrality) [inline examples/substrate/use-futurepass/src/registerDelegate.ts:28]: need the `export DELEGATE_PRIVATE_KEY=0x000` in the `README.md` file

**Review by ken-centrality — APPROVED** [review]

**Pull request merged**